src/ray/design_docs/id_specification.md

```diff
@@ -54,3 +54,6 @@ An `ObjectID` contains 2 parts:
   1 <= idx <= num_return_objects is reserved for the task's return objects, while
   idx > num_return_objects is available for the task's put objects.
 - `TaskID`: 24 bytes to indicate the ID of the task to which this object belongs.
+  Note: For `ray.put()` IDs only, the first byte of the `TaskID` is zeroed out
```

Comment:
To clarify, does this also apply to the return ids? Per the description of the PR: `deterministically generate the same IDs for any ray.put or .remote task calls`

Reply:
Ah I'll clarify this. It does not apply to return IDs, but all child task IDs are now seeded with the execution number of the parent task ID, so this will prevent collisions for the return IDs.